agot-bg-game-server/src/client/game-state-panel/SelectRegionComponent.tsx

```diff
@@ -79,5 +79,6 @@ export default class SelectRegionComponent extends Component<GameStateComponentP
         }
 
         this.props.gameState.select(this.selectedRegion);
+        this.selectedRegion = null;
```

Comment:
It's a bit of a hack.

When the retreat region is chosen, the server sends a `move-units` packet to the client before sending a packet to change the game state. The client is thus in a weird state where the retreating unit has retreated, but the GameState is still a ResolveRetreatGameState, which causes an error because the UI attemps to compute the number of casualties (which assumes that the loser army is still in the attacking region or the defending region).

More situations like this will probably arise later in the development. A big fix for this would be to make sure all modifications done server-side are sent at one to the client.